common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -200,6 +201,35 @@ def get(request):
     return request.session.get('partial_pipeline')
 
 
+def get_real_social_auth_object(request):
+    """
+    At times, the pipeline will have a "social" kwarg that contains a dictionary
+    rather than an actual DB-backed UserSocialAuth object. We need the real thing,
+    so this method allows us to get that by passing in the relevant request.
+    """
+    running_pipeline = get(request)
+    if running_pipeline and 'social' in running_pipeline['kwargs']:
+        social = running_pipeline['kwargs']['social']
+        if isinstance(social, dict):
+            social = UserSocialAuth.objects.get(uid=social.get('uid', ''))
```

Comment:
We should raise a UserSocialAuthDoesNotExist error, which will cause a 500 error. We can come up with a different error to raise if you'd prefer.